### PR TITLE
Rename EventStream to SourceStream

### PR DESCRIPTION
**Pull request by Shane32**
Partial towards #3022

I'm okay with these changes, but I am biased because I do not yet use subscriptions.  I do not know how much this would affect other users.

Also, I did not yet rename `ISourceStreamResolver.SubscribeAsync`.  Personally I am much more concerned about that name, as it is inaccurate (subscribing does not yet occur) than whether it is "event stream" or "source stream".  I suggest `ResolveAsync` which matches `IFieldResolver.ResolveAsync`.  Either way you are resolving something - either an object or the source stream.  But you are not subscribing.  So we can do that in another PR if you like.

Our changes for v5 as it relates to subscriptions have pretty much been overhauled completely, so now's the time to change it if we want to.

### CODE REVIEW

**Comment by sungam3r** [comment]
Agree regarding renaming `SubscribeAsync` to `ResolveAsync`. In

File: src/GraphQL/Types/Fields/FieldType.cs [hunk 1]
```diff
@@ -79,6 +79,6 @@ internal void SetName(string name, bool validate)
         /// <summary>
         /// Gets or sets a subscription resolver for the field. Only applicable to subscription fields.
         /// </summary>
-        public IEventStreamResolver? Subscriber { get; set; }
+        public ISourceStreamResolver? Subscriber { get; set; }
```
Comment (sungam3r) [inline src/GraphQL/Types/Fields/FieldType.cs:82]:
```suggestion
        public ISourceStreamResolver? StreamResolver { get; set; }
```

File: src/GraphQL/Utilities/FieldConfig.cs [hunk 2]
```diff
@@ -50,7 +50,7 @@ public FieldConfig(string name)
         /// <summary>
         /// Gets or sets the event stream resolver.
         /// </summary>
-        public IEventStreamResolver? Subscriber { get; set; }
+        public ISourceStreamResolver? Subscriber { get; set; }
```
Comment (sungam3r) [inline src/GraphQL/Utilities/FieldConfig.cs:53]:
```suggestion
        public ISourceStreamResolver? StreamResolver { get; set; }
```

File: src/GraphQL.MicrosoftDI/DynamicScopedEventStreamResolver.cs [hunk 3]
```diff
@@ -6,17 +6,17 @@ namespace GraphQL.MicrosoftDI
     /// <summary>
     /// When resolving a field, this implementation calls
     /// <see cref="IResolveFieldContext.RequestServices"/>.<see cref="ServiceProviderServiceExtensions.CreateScope(IServiceProvider)">CreateScope</see>
-    /// to create a dependency injection scope. Then it calls the specified <see cref="IEventStreamResolver"/>, passing the scoped service provider
+    /// to create a dependency injection scope. Then it calls the specified <see cref="ISourceStreamResolver"/>, passing the scoped service provider
     /// within <see cref="IResolveFieldContext.RequestServices"/>, and returns the result.
     /// </summary>
-    internal class DynamicScopedEventStreamResolver : IEventStreamResolver
+    internal class DynamicScopedSourceStreamResolver : ISourceStreamResolver
```
Comment (sungam3r) [inline src/GraphQL.MicrosoftDI/DynamicScopedEventStreamResolver.cs:12]: You forgot to rename file.

**Comment by sungam3r** [comment]
Now it is easier to understand which code for what is responsible by reading the specification.

File: src/GraphQL/Resolvers/ISourceStreamResolver.cs [hunk 4]
```diff
@@ -6,9 +6,9 @@ namespace GraphQL.Resolvers
     /// which will resolve child fields using the event data as the source. Then the resolved graph is
     /// returned to the client, and the process repeats for further event notifications. 
     /// </summary>
-    public interface IEventStreamResolver
+    public interface ISourceStreamResolver
     {
-        /// <inheritdoc cref="IEventStreamResolver"/>
+        /// <inheritdoc cref="ISourceStreamResolver"/>
         ValueTask<IObservable<object?>> SubscribeAsync(IResolveFieldContext context);
```
[the remark this answers is not in the document]
Comment (Shane32) [reply src/GraphQL/Resolvers/ISourceStreamResolver.cs:12]: Not in this PR.

File: src/GraphQL/Utilities/FieldConfig.cs (hunk 2 above)
Comment (Shane32) [reply src/GraphQL/Utilities/FieldConfig.cs:53]: Not in this PR

**Comment by Shane32** [comment]
Next PR: SubscribeAsync -> ResolveAsync  and  Subscriber -> StreamResolver

**Review by sungam3r — APPROVED** [review]

**Pull request merged**